StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/UsingCodeFixProvider.cs

```diff
@@ -0,0 +1,207 @@
+                SA1209UsingAliasDirectivesMustBePlacedAfterOtherUsingDirectives.DiagnosticId,
+                SA1210UsingDirectivesMustBeOrderedAlphabeticallyByNamespace.DiagnosticId,
+                SA1211UsingAliasDirectivesMustBeOrderedAlphabeticallyByAliasName.DiagnosticId);
+
+        /// <inheritdoc/>
+        public override ImmutableArray<string> FixableDiagnosticIds => FixableDiagnostics;
+
+        /// <inheritdoc/>
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            foreach (Diagnostic diagnostic in context.Diagnostics.Where(d => FixableDiagnostics.Contains(d.Id)))
+            {
+                context.RegisterCodeFix(new PreservingDocumentChangeAction("Reorder using statements", token => GetTransformedDocumentAsync(context.Document, diagnostic, token)), diagnostic);
+            }
+
+            return Task.FromResult(true);
+        }
+
+        private static async Task<Document> GetTransformedDocumentAsync(Document document, Diagnostic diagnostic, CancellationToken token)
+        {
+            var syntaxRoot = await document.GetSyntaxRootAsync(token).ConfigureAwait(false);
+
+            var usingSyntaxRewriter = new UsingSyntaxRewriter();
+            var newSyntaxRoot = usingSyntaxRewriter.Visit(syntaxRoot);
+            var indentationOptions = IndentationOptions.FromDocument(document);
+
+            newSyntaxRoot = ProcessUsingDirectives(newSyntaxRoot, usingSyntaxRewriter.UsingDirectives, indentationOptions);
+            var newDocument = document.WithSyntaxRoot(newSyntaxRoot);
+
+            return newDocument;
+        }
+
+        private static SyntaxNode ProcessUsingDirectives(SyntaxNode newSyntaxRoot, ImmutableArray<UsingDirectiveSyntax> usingDirectives, IndentationOptions indentationOptions)
+        {
+            var systemUsings = new List<UsingDirectiveSyntax>();
+            var otherUsings = new List<UsingDirectiveSyntax>();
+            var aliases = new List<UsingDirectiveSyntax>();
+            var staticImports = new List<UsingDirectiveSyntax>();
+
+            foreach (var usingDirective in usingDirectives)
+            {
+                if (usingDirective.Alias != null)
+                {
+                    aliases.Add(usingDirective);
+                }
+                else if (!usingDirective.StaticKeyword.IsKind(SyntaxKind.None))
+                {
+                    staticImports.Add(usingDirective);
+                }
+                else
+                {
+                    var firstPart = usingDirective.Name;
+
+                    while (firstPart is QualifiedNameSyntax)
+                    {
+                        firstPart = ((QualifiedNameSyntax)firstPart).Left;
+                    }
+
+                    var firstPartName = ((IdentifierNameSyntax)firstPart).Identifier.ValueText;
+
+                    if (string.CompareOrdinal(firstPartName, "System") == 0)
```

Comment:
You could use `"System".Equals(firstPartName, StringComparison.Ordinal)` her instead